jbmc/src/java_bytecode/ci_lazy_methods.cpp

```diff
@@ -246,8 +252,12 @@ bool ci_lazy_methodst::handle_virtual_methods_with_no_callees(
     // Add the call class to instantiated_classes and assert that it
     // didn't already exist
     const irep_idt &call_class = virtual_function_call.get(ID_C_class);
-    auto ret_class = instantiated_classes.insert(call_class);
-    CHECK_RETURN(ret_class.second);
+    bool added_class = instantiated_classes.count(call_class) == 0;
+    lazy_methods_loader.add_all_needed_classes(
+      to_pointer_type(
+        to_java_method_type(virtual_function_call.type()).get_this()->type()));
+
+    CHECK_RETURN(added_class);
```

Comment:
Surely this CHECK_RETURN should be right below the call to `instantiated_classes.count`? Is it also possible to check after calling `add_all_needed_classes` that `call_class` is in `instantiated_classes`?

Reply:
In regards to the former - I'll move it. The latter isn't really doable because it was previously checking to make sure that we're inserting the class, and just checking it exists afterwards would fail in the situation that it already exists before we even make the call to add.

Reply:
I meant checking it before and afterwards